### ISSUE TITLE
GitPoller is broken in 3.11.3, 3.11.2 works

### ISSUE THREAD

**Issue by kayhayen** [buildbot/buildbot#7694]
I followed the deprecation warning and changed the branch argument, hoping it helps, it did not.

I have this change source:

```
c["change_source"].append(
    GitPoller(
        "git@localhost:Py2C.git",
        workdir="gitpoller-private",
        branches=["main"],
        project="private",
        pollInterval=30,
    )
)
```

later

```
c["schedulers"].append(
    SingleBranchScheduler(
        name="run-tests",
        change_filter=filter.ChangeFilter(
            branch="main",
            project="private",
        ),
        treeStableTimer=tree_timer_min,
        builderNames=test_cases,
        fileIsImportant=isImportantForTests,
    )
)
```

Seems like nothing special. The changelog of 3.11.3 mentions GitPoller a lot. I have had no changes recognized in the 9 days it has been running. I dropped my 9GB state file in an attempt to maybe remove broken state and started all over. I had 8 years old pending build requests still, probably a good idea.

Only the downgrade has helped.

**Comment by yan12125** [comment]
The fix is on git master branch and should be included in the next version: https://github.com/buildbot/buildbot/issues/7668

**Comment by mlabbe** [comment]
Just a quick note to say I lost some good debugging time to this one.  Fix was to downgrade to 3.11.2. it would be great if the manual was patched to note this around the Gitpoller section in the interim. :)

**Comment by mokibit** [comment]
@p12tic I believe this issue can be closed as the problem was fixed by backporting in PR https://github.com/buildbot/buildbot/pull/7691.
